src/openforms/analytics_tools/tests/test_analytics_rendering.py

```diff
@@ -0,0 +1,122 @@
+from django.urls import reverse
+
+from cookie_consent.models import CookieGroup
+from django_webtest import WebTest
+
+from openforms.analytics_tools.models import AnalyticsToolsConfiguration
+from openforms.forms.tests.factories import FormFactory
+
+
+class AnalyticsToolsRenderingTest(WebTest):
+    """Integration tests for rendering of analytics snippets.
+
+    The tests check that the analytics snippets are rendered if they are
+    enabled (by the admin) and cookies are accepted (by the user). They
+    do *not* test whether the scripts are properly executed, which requires
+    either manual testing or functional tests with Selenium.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        form = FormFactory.create()
+        cls.url = reverse("forms:form-detail", kwargs={"slug": form.slug})
+        config = AnalyticsToolsConfiguration.get_solo()
+        config.analytics_cookie_consent_group = CookieGroup.objects.get(
+            varname="analytical"
+        )
+        config.save()
+        cls.config = config
+
+    def test_google_analytics_rendering(self):
+        """Assert that the Google Analytics script is rendered"""
+
+        # Enable Google analytics
+        self.config.enable_google_analytics = True
+        self.config.gtm_code = "GTM-XXXX"
+        self.config.ga_code = "UA-XXXXX-Y"
+        self.config.save()
+
+        # Accept cookies
+        form_page = self.app.get(self.url)
+        accept_form = form_page.forms[0]
+        refreshed_form_page = accept_form.submit().follow()
+
+        soup = refreshed_form_page.html
+        script = soup.find("script")
+        script_text = script.get_text()
+
+        self.assertIn("googletagmanager", script_text)
```

Comment:
to answer your question in the comment - let's give every analytics script an id (e.g. `<script id="analytics-matomo">` and test for that by finding the node with this ID and asserting that it's a script tag. That should be sufficient.